### ISSUE TITLE
Polyfill.io breaks legacy polyfill in IE11 

### ISSUE THREAD

**Issue by lucasavila00** [Financial-Times/polyfill-service#2751]
### Describe the bug

I don't know if I should report this bug to Polyfill.io or Vite. Please pardon me if this should not have been written here.

I'm building a website that needs to run on IE11. I'm using `@vitejs/plugin-legacy` for it.

If I load the Intersection Observer polyfill from polyfill.io, the legacy code Vite built doesn't work. (The error "System is undefined" seems to mean SytemJS doesn't work, and the client-side app doesn't load)

![image](https://github.com/Financial-Times/polyfill-service/assets/12750442/add04b4f-054c-43a0-9cb7-c2d236d9b1a5)

I tried searching around and debugging it but this is as much information as I could get.

On the reproduction, the important parts are:

- Loading the polyfill https://github.com/lucasavila00/vite-polyfill-ie11/blob/main/index.html#L8
- The polyfill https://github.com/lucasavila00/vite-polyfill-ie11/blob/main/public/polyfill.js
- Vite legacy config https://github.com/lucasavila00/vite-polyfill-ie11/blob/main/vite.config.ts#L8
- Async Vite bundle https://github.com/lucasavila00/vite-polyfill-ie11/blob/main/index.html#L12

The use-case of this website requires external JS to be loaded into my website.

I have verified that this Intersection Observer polyfill does not break Vite code: https://github.com/GoogleChromeLabs/intersection-observer but ideally my website should not break if external - non-adversary - JS is loaded.

### Reproduction

https://github.com/lucasavila00/vite-polyfill-ie11

### Steps to reproduce

`npm install` && `npm run build` && `cd dist` && `python -m http.server`

### System Info

```shell
System:
    OS: Linux 6.2 Pop!_OS 22.04 LTS
    CPU: (16) x64 AMD Ryzen 7 3800X 8-Core Processor
    Memory: 44.64 GB / 62.72 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.10 - ~/.npm-global/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 112.0
  npmPackages:
    @vitejs/plugin-legacy: ^4.0.4 => 4.0.4 
    vite: ^4.3.9 => 4.3.9
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lucasavila00** [comment]
Also reported to Vite https://github.com/vitejs/vite/issues/13494

**Comment by JakeChampion** [comment]
Regarding SystemJS, it needs polyfills loaded in order to work on IE11 -- https://github.com/systemjs/systemjs#ie11-support

Do you know if those polyfills are loaded before SystemJS runs?

I also noticed the polyfill.io code is inlined, but it looks to not put the polyfills behind feature-detection, that will cause issues for lots of browsers, I recommend using this bundle -- https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserver,IntersectionObserverEntry&flags=gated,always

**Comment by lucasavila00** [comment]
> Do you know if those polyfills are loaded before SystemJS runs?

Yes, they are.

If I remove the polyfill.io polyfill from the project, the website loads correctly:

![image](https://github.com/Financial-Times/polyfill-service/assets/12750442/78eb6886-d28d-489e-ac23-03f0ead97a91)


> I also noticed the polyfill.io code is inlined, but it looks to not put the polyfills behind feature-detection, that will cause issues for lots of browsers, I recommend using this bundle 

I inlined the version I got on IE11. IRL the polyfill from polyfill.io comes from a customer/user of my website, it is not under my control.

**Comment by JakeChampion** [comment]
This is a confusing issue, are you using polyfill.io in your project?
It sounds like you are not using polyfill.io and are using `@vitejs/plugin-legacy` -- is that correct?

**Comment by lucasavila00** [comment]
>  It sounds like you are not using polyfill.io and are using @vitejs/plugin-legacy -- is that correct?

Yes, it is correct. But my website supports the user/consumer adding custom JS script (ads code). A user/consumer of my website added the polyfill.io code and it broke.

The sample project https://github.com/lucasavila00/vite-polyfill-ie11 is just a simple reproduction of the HTML that eventually is shipped by my website. (IRL it'd be SSR'd and not SPA'd, etc)

**Comment by lucasavila00** [comment]
I was able to track it down to just `Object.defineProperty`. https://polyfill.io/v3/polyfill.js?features=Object.defineProperty&flags=always will break Vite's legacy code.
